MovieMVP/ServiceLayer/NetworkService.swift

```diff
@@ -0,0 +1,121 @@
+
+    private enum Constants {
+        static let apiKeyQueryText = "api_key=8216e974d625f2a458a739c20007dcd6"
+        static let languageQueryText = "&language=ru-RU"
+        static let pageQueryText = "&page=1"
+        static let themoviedbQueryText = "https://api.themoviedb.org/3/movie/"
+        static let posterPathQueryText = "https://image.tmdb.org/t/p/w500"
+        static let similarQueryText = "/similar?"
+        static let resultsText = "results"
+        static let posterPathText = "poster_path"
+        static let idText = "id"
+        static let overviewText = "overview"
+        static let releaseDateText = "release_date"
+        static let titleText = "title"
+        static let voteAverageText = "vote_average"
+        static let voteCountText = "vote_count"
+    }
+
+    // MARK: - Public Methods
+
+    func fetchMovies(categoryMovies: String, completion: @escaping (Result<[Movie]?, Error>) -> Void) {
+        let urlString = "\(Constants.themoviedbQueryText)\(categoryMovies)\(Constants.apiKeyQueryText)" +
+            "\(Constants.languageQueryText)\(Constants.pageQueryText)"
+        guard let url = URL(string: urlString) else { return }
+
+        AF.request(url, method: .get).responseData { response in
+            guard let safeData = response.value
+            else {
+                guard let safeError = response.error else { return }
+                completion(.failure(safeError))
+                return
+            }
+            var movies: [Movie]? = []
+            if let json = try? JSON(data: safeData) {
+                for result in json[Constants.resultsText].arrayValue {
+                    let id = result[Constants.idText].intValue
+                    let overview = result[Constants.overviewText].stringValue
+                    let posterPath = result[Constants.posterPathText].stringValue
+                    let releaseDate = result[Constants.releaseDateText].stringValue
+                    let title = result[Constants.titleText].stringValue
+                    let voteAverage = result[Constants.voteAverageText].doubleValue
+                    let voteCount = result[Constants.voteCountText].doubleValue
+                    movies?.append(Movie(
+                        id: id,
+                        overview: overview,
+                        posterPath: posterPath,
+                        releaseDate: releaseDate,
+                        title: title,
+                        voteAverage: voteAverage,
+                        voteCount: voteCount
+                    ))
+                }
+            }
+            completion(.success(movies))
+        }
+    }
+
+    func setupImageFromURLImage(posterPath: String, completion: @escaping (Result<Data?, Error>) -> Void) {
+        guard let imageMovieNameURL =
+            URL(string: "\(Constants.posterPathQueryText)\(posterPath)")
+        else { return }
+        AF.request(imageMovieNameURL, method: .get).responseData { response in
```

Comment:
```suggestion
        AF.request(imageMovieNameURL).responseData { response in
```
необязательный параметр